File: core/src/main/java/com/illuzionzstudios/tab/listener/LegacyBlocker.java
/**
 * Copyright © 2020 Property of Illuzionz Studios, LLC
 * All rights reserved. No part of this publication may be reproduced, distributed, or
 * transmitted in any form or by any means, including photocopying, recording, or other
 * electronic or mechanical methods, without the prior written permission of the publisher,
 * except in the case of brief quotations embodied in critical reviews and certain other
 * noncommercial uses permitted by copyright law. Any licensing of this software overrides
 * this statement.
 */
package com.illuzionzstudios.tab.listener;

import com.comphenix.protocol.PacketType;
import com.comphenix.protocol.events.PacketAdapter;
import com.comphenix.protocol.events.PacketEvent;
import org.bukkit.plugin.Plugin;

/**
 * Used for players joining on outdated clients
 */
public class LegacyBlocker extends PacketAdapter {

    public LegacyBlocker(Plugin plugin) {
        super(plugin, PacketType.Play.Server.PLAYER_INFO);
    }

    @Override
    public void onPacketSending(PacketEvent event) {
        event.setCancelled(true);
    }
}

File: core/src/main/kotlin/com/illuzionzstudios/tab/tab/LegacyBlocker.kt
package com.illuzionzstudios.tab.tab

import com.comphenix.protocol.PacketType
import com.comphenix.protocol.events.PacketAdapter
import com.comphenix.protocol.events.PacketEvent
import com.illuzionzstudios.mist.plugin.SpigotPlugin

class LegacyBlocker(plugin: SpigotPlugin): PacketAdapter(plugin, PacketType.Play.Server.PLAYER_INFO) {

    override fun onPacketSending(event: PacketEvent) {
        event.isCancelled = true
    }

}